### PR TITLE
Restructure docs

### PR DESCRIPTION
**Pull request by hofaflo**
- While the recursive autosummary we currently use to populate the API reference makes sure we can't forget to add a new function to the docs, it doesn't reflect the actual public API, since it follows the module structure of the package. Listing the public functions and classes manually in `doc/source/api.rst` resolves this issue.
- The doc index page now shows an overview of the doc pages instead of the main readme.
- The main readme's part on heartbeat detection is now a separate page in the docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ SleepECG provides tools for sleep stage classification when [EEG](https://en.wik
 - classifying sleep stages (which includes the complete preprocessing, feature extraction, and classification pipeline) (*TODO*).
 
 
+## Documentation
+Documentation for SleepECG is available on [readthedocs](https://sleepecg.readthedocs.io/en/stable/index.html).
```
Comment (cbrnr) [inline README.md:14]:
```suggestion
Documentation for SleepECG is available on [Read the Docs](https://sleepecg.readthedocs.io/en/stable/index.html).
```

File: doc/source/api.rst [hunk 2]
```diff
@@ -1,13 +1,60 @@
...
-   DO NOT DELETE THIS FILE! It contains the all-important `.. autosummary::` directive with `:recursive:` option, without
-   which API documentation wouldn't get extracted from docstrings by the `sphinx.ext.autosummary` engine. It is hidden
-   (not declared in any toctree) to remove an unnecessary intermediate page; index.rst instead points directly to the
-   package page. DO NOT REMOVE THIS FILE!
+SleepECG API Reference
+======================
+
+.. automodule:: sleepecg
+
+This page gives an overview of all public SleepECG functions and classes.
```
Comment (cbrnr) [inline doc/source/api.rst:6]:
```suggestion
This page lists all public SleepECG functions and classes.
```

File: doc/source/index.md [hunk 3]
```diff
@@ -2,11 +2,19 @@
    :hidden:
 
    Home Page <self>
```
Comment (cbrnr) [inline doc/source/index.md:4]:
```suggestion
   Home <self>
```

File: doc/source/index.md [hunk 4]
```diff
@@ -2,11 +2,19 @@
...
-   API reference <generated/sleepecg>
+   API reference <api>
```
Comment (cbrnr) [inline doc/source/index.md:5]:
```suggestion
   API <api>
```

File: doc/source/index.md [hunk 5]
```diff
@@ -2,11 +2,19 @@
...
    Feature Extraction <feature_extraction>
    Configuration <configuration>
 ```
 
-```{include} ../../README.md
-```
+# SleepECG
+Welcome! This is the documentation for version **{{ version }}** of SleepECG, a toolbox for sleep stage classification using ECG data.
+
+- [**API reference**](./api) has detailed information about all public functions and classes in SleepECG.
```
Comment (cbrnr) [inline doc/source/index.md:15]:
```suggestion
- [**API**](./api) has detailed information about all public functions and classes in SleepECG.
```

File: doc/source/index.md [hunk 6]
```diff
@@ -2,11 +2,19 @@
...
+- [**Datasets**](./datasets) shows all avaiable datasets and instructions about retrieving NSRR data.
+- [**Heartbeat Detection**](./heartbeat_detection) demonstrates how to use the included heartbeat detector and shows benchmark results.
+- [**Feature Extraction**](./feature_extraction) lists the implemented HRV features.
+- [**Configuration**](./configuration) provides explains the possible configuration settings in SleepECG.
```
Comment (cbrnr) [inline doc/source/index.md:19]:
```suggestion
- [**Configuration**](./configuration) explains the possible configuration settings in SleepECG.
```

File: doc/source/index.md [hunk 7]
```diff
@@ -2,11 +2,19 @@
...
+   Heartbeat Detection <heartbeat_detection>
    Feature Extraction <feature_extraction>
    Configuration <configuration>
 ```
 
-```{include} ../../README.md
-```
+# SleepECG
+Welcome! This is the documentation for version **{{ version }}** of SleepECG, a toolbox for sleep stage classification using ECG data.
+
```
Comment (cbrnr) [inline doc/source/index.md:14]: It's probably worth inserting the installation instructions here. Then I would also link to the contributing guide and changelog right here, and only after that I'd list the other items of the website (API, Datasets, ...).

**Review by cbrnr — COMMENTED** [review]
This is a nice structure! I'd only change the homepage a little bit and include more information (see my comment).

**Comment by hofaflo** [comment]
Done, thanks for the suggestions!

**Comment by cbrnr** [comment]
Thanks @hofaflo!

**Pull request merged**